Proposed changes to build_b_call()

jimrothstein opened the pull request:
@ARawles

jimrothstein committed:
trying things with jim_build version of build.R

jimrothstein committed:
proposed changes to build_b_call(), see NEWS.md

ARawles commented:
The build changes look great!

Would you be able to either formalise or remove the temporary tests, and finalise the changes in the NEWS.md so that they match the rest of the release notes? After that, I'd be happy to integrate the changes.

Don't worry about the Travis build failing - it's just because I wrote the test cases to not use URL encoding of special characters, but I'm happy to fix that myself.

jimrothstein committed:
Proposed BMRSr 1.0.1.9001

jimrothstein commented:
@ARawles 
Bumped version to BMRSr **1.0.1.9001**.
Not quite satisfied with tests but want to think about it ..... helpful  to me to have working example around.

ARawles committed:
Removed API key stored at environmental variable level

Having the API key as an environmental variable makes sense when you're testing interactively but it's going to cause the tests to fail on other machines as they won't have that variable.
Given that we're not actually making a request - we're just building the URL - we can just use a fake API key for these tests.

ARawles reviewed (approved):
Happy with the changes as they currently stand, and I'm happy to fix the test failures.

Thanks for your help @jimrothstein
